ruby/example_code/lambda/lambda_basics.rb

```diff
@@ -0,0 +1,295 @@
+#
+# Shows how to use the AWS SDK for Ruby v3 to manage and invoke AWS Lambda
+# functions.
+
+require "aws-sdk-lambda"
+require "aws-sdk-s3"
+require "aws-sdk-iam"
+require "logger"
+require "json"
+require "zip"
+
+$cloudwatch_client = Aws::CloudWatchLogs::Client.new
+$iam_client = Aws::IAM::Client.new
+
+# snippet-start:[ruby.example_code.ruby.LambdaWrapper.full]
+# snippet-start:[ruby.example_code.ruby.LambdaWrapper.decl]
+class LambdaWrapper
+  attr_accessor :lambda_client
+
+  def initialize
+    @lambda_client = Aws::Lambda::Client.new
+    @logger = Logger.new($stdout)
+    @logger.level = Logger::WARN
+  end
+  # snippet-end:[ruby.example_code.ruby.LambdaWrapper.decl]
+
+  # snippet-start:[ruby.example_code.lambda.setup_iam]
+  # Get an AWS Identity and Access Management (IAM) role.
+  #
+  # @param iam_role_name: The name of the role to retrieve.
+  # @param action: whether to create or destroy the IAM apparatus
+  # @return: The IAM role.
+  def manage_iam(iam_role_name, action)
+    role_policy = {
+      'Version': "2012-10-17",
+      'Statement': [
+        {
+          'Effect': "Allow",
+          'Principal': {
+            'Service': "lambda.amazonaws.com"
+          },
+          'Action': "sts:AssumeRole"
+        }
+      ]
+    }
+    case action
+    when "create"
+      role = $iam_client.create_role(
+        role_name: iam_role_name,
+        assume_role_policy_document: role_policy.to_json
+      )
+      $iam_client.attach_role_policy(
+        {
+          policy_arn: "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+          role_name: iam_role_name
+        }
+      )
+      $iam_client.wait_until(:role_exists, { role_name: iam_role_name }) do |w|
+        w.max_attempts = 5
+        w.delay = 5
+      end
+      @logger.debug("Successfully created IAM role: #{role['role']['arn']}")
+      @logger.debug("Enforcing a 10-second sleep to allow IAM role to activate fully.")
+      sleep(10)
+      return role, role_policy.to_json
+    when "destroy"
+      $iam_client.detach_role_policy(
+        {
+          policy_arn: "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+          role_name: iam_role_name
+        }
+      )
+      $iam_client.delete_role(
+        role_name: iam_role_name
+      )
+      @logger.debug("Detached policy & deleted IAM role: #{iam_role_name}")
+    else
+      raise "Incorrect action provided. Must provide 'create' or 'destroy'"
+    end
+  rescue Aws::Lambda::Errors::ServiceException => e
+    @logger.error("There was an error creating role or attaching policy:\n #{e.message}")
+  end
+  # snippet-end:[ruby.example_code.lambda.setup_iam]
+
+  # snippet-start:[ruby.example_code.lambda.create_deployment_package]
+  # Creates a Lambda deployment package in .zip format
+  # This zip can be passed directly as a string to Lambda when creating the function.
+  #
+  # @param source_file: The name of the object without suffix for Lambda file and zip.
```

Comment:
@param source_file: The name of the object, without suffix, for the Lambda file and zip.

Reply:
Done.